integrated_channels/canvas/client.py

```diff
@@ -78,22 +91,62 @@ def course_update_endpoint(canvas_base_url, course_id):
         )
 
     def create_content_metadata(self, serialized_data):
+        """
+        Creates a course in Canvas.
+        If course is not found ,easy!  create it as usual
+        If course found, it will be one of these workflow_states per doc /the current state
+            of the course one of 'unpublished', 'available', 'completed', or 'deleted'
+                If available: issue an update with latest field values
+                If completed: this happens if a course has been concluded.
+                    update it to change status to offer as per course[event]=offer
+                If ‘unpublished’ - still just update
+                If ‘deleted’ - take no action for now.
+        https://canvas.instructure.com/doc/api/courses.html#method.courses.update
+        """
         self._create_session()
 
-        # step 1: create the course
-        status_code, response_text = self._post(self.course_create_url, serialized_data)
-        created_course_id = json.loads(response_text)['id']
-
-        # step 2: upload image_url and any other details
-        self._update_course_details(created_course_id, serialized_data)
+        course_details = json.loads(serialized_data.decode('utf-8'))['course']
+        edx_course_id = course_details['integration_id']
+        located_course = self._find_course_by_course_id(edx_course_id)
+
+        if not located_course:
+            # Course does not exist: Create the course
+            status_code, response_text = self._post(self.course_create_url, serialized_data)
+            created_course_id = json.loads(response_text)['id']
+
+            # step 2: upload image_url and any other details
+            self._update_course_details(created_course_id, course_details)
+        else:
+            workflow_state = located_course['workflow_state']
+            if workflow_state.lower() == 'deleted':
+                LOGGER.warning(
```

Comment:
Can you please use the new generate_formatted_log function for new logs in this file? user id will be none, but other items can be filled in.

Reply:
Also can you raise this to "ERROR" ? If we have this it's going to cause sync problems for users, we want to know about it in the logs and possibly alert on it.

Reply:
both make sense! will do